Assets/Scripts/Model/NodeClass.cs

```diff
@@ -1,27 +1,25 @@
 ﻿using System;
 using System.Runtime.Serialization;
 using Newtonsoft.Json;
-using Newtonsoft.Json.Converters;
 using UnityEngine;
 
 namespace Assets.Scripts.Model
 {
   [Serializable]
   public class NodeClass
   {
-    public int id;
-    public string name;
-    [JsonConverter(typeof(StringEnumConverter))]
-    public PrimitiveType? shape;
-    public string texturePath;
-    public float? scale;
+    [JsonProperty] public int Id { get; private set; }
+    [JsonProperty] public string Name { get; private set; }
+    [JsonProperty] public PrimitiveType? Shape { get; private set; }
+    [JsonProperty] public string TexturePath { get; private set; }
+    [JsonProperty] public float? Scale { get; private set; }
 
 
     [OnDeserialized]
     public void OnDeserialized(StreamingContext context)
     {
-      if (string.IsNullOrEmpty(name))
-        name = id.ToString();
+      if (string.IsNullOrEmpty(Name))
+        Name = Id.ToString();
```

Comment:
is it ok style for c# to make property provide default value?
like 
string _name
Name { get => if (_name == "")
and so on?

Reply:
Yeah, I don't see why not. Though I would rather keep the properties as "pure" for easy serializing/deserializing.